src/power_hand.c

```diff
@@ -201,10 +200,32 @@ long can_thing_be_picked_up_by_player(const struct Thing *thing, PlayerNumber pl
     return can_cast_spell(plyr_idx, PwrK_HAND, thing->mappos.x.stl.num, thing->mappos.y.stl.num, thing, CastChk_Default);
 }
 
-long can_thing_be_picked_up2_by_player(const struct Thing *thing, PlayerNumber plyr_idx)
+TbBool can_thing_be_picked_up2_by_player(const struct Thing *thing, PlayerNumber plyr_idx)
 {
     //TODO: rewrite, then give it better name
-    return _DK_can_thing_be_picked_up2_by_player(thing, plyr_idx);
+    // return _DK_can_thing_be_picked_up2_by_player(thing, plyr_idx);
+    if ( thing->class_id != TCls_Creature )
+    {
+        return ( (thing->class_id == TCls_Object) && (object_is_pickable_by_hand_for_use(thing, plyr_idx)) );
+    }
+    if ( (game.armageddon_cast_turn > 0) && ( ((game.armageddon.count_down + game.armageddon_cast_turn) <= game.play_gameturn) ) )
```

Comment:
I'm pretty sure you have an incorrect number of () here.
I think you want it to be:
```c
if ( (game.armageddon_cast_turn > 0) && (
(game.armageddon.count_down + game.armageddon_cast_turn) <= game.play_gameturn) ) 
```